### PR TITLE
Replaced `self.data["attr"]` and `self.uniforms["attr"]` with `self.attr`.

### LINKED ISSUE #1883: Use `self.attr` as opposed to `self.data["attr"]` for OpenGL files.
## Enhancement proposal

Throughout the codebase for opengl files: `opengl_mobject.py` `svg_path.py`, `opengl_surface.py`, `opengl_vectorized_mobject.py`, `opengl_renderer.py`. They use a dict to store some information such as `points`, when it could just be accessed as an attribute instead (like how it's done in the regular mobjects).

This is probably a relic from porting over from 3b1b and should be changed to how someone would normally expect attributes to work.

### PR DESCRIPTION
**Pull request by hydrobeam**
## Motivation and Explanation: Why and how do your changes improve the library?

closes #1883 + addresses `self.uniforms["attr"]` since it's implemented in a similar way

### Reasoning

@ryanmccauley211 mentioned that attributes still needed to be grouped as there were several methods that acted on the grouped `self.data` + `self.uniforms`. The solution here is to create a `_Data` and `_Uniforms` descriptor that is stored in `utils.config_ops.py`.

These are implemented similarily: they access the attribute via `self.data["attr"]` and write to it the same way. All the regular operations you'd carry out on the attributes work as expected.

### Code Changes
Direct calls such as `self.data["attr"]` were replaced with `self.attr`; however, generic `self.data[key]` calls were not touched. The relevant `_Data`/ `_Uniform` variables are now explicitly set as class variables 

Notable file specifc changes:

#### `opengl_mobject.py` 
- used `getattr()` for data/uniforms since these are defined in `opengl_vmobject`/`opengl_surface` and shouldn't be overwritten.
- removed `init_uniforms` as it was made redundant.

#### `opengl_vectorized_mobject.py`
- added `self.data = {}` to store the `stroke_width` _Data variable, without it, it would try to set to a `data` that does not yet exist as `super()` was not called..
- removed `stroke_width` from `init_data` as it was overwriting the stroke_width value passed. A little akward since `stroke_width` is now defined in `init_colors`... but it's fine imo.

#### `opengl_renderer.py` 
- had to move the definition of `euler_angles` of `OpenGLCamera` after the super() and call `refresh_rotation_matrix` again since the scene relies on a variable instantiated in that method to render.

#### `opengl_surface.py`
- same `self.data = {}` logic as before,  also removed `init_uniforms`.

#### `number_line.py` 
- self.stroke_width was being defined and raising an error since `data` had not been created via super yet.
    - removed this as it was just being overwritten in `super()` anyways. 

Here's some code to see that adjusting the attribute manually also adjusts it within `self.data` (+ the test I included):

```py
class TestData(Scene):
    def construct(self):
        a = Square().move_to(RIGHT)
        print(f"first data: {a.data['bounding_box']}")
        # first data: [[ 0. -1.  0.]
        # [ 1.  0.  0.]
        # [ 2.  1.  0.]]

        print(f"first b_b: {a.bounding_box}")
        # first b_b: [[ 0. -1.  0.]
        # [ 1.  0.  0.]
        # [ 2.  1.  0.]]

        print(f"b_b index[1]: {a.bounding_box[1]}")
        # b_b index[1]: [1. 0. 0.]
        a.bounding_box[1] = 300

        print(f"second b_b: {a.bounding_box}")
        # second b_b: [[  0.  -1.   0.]
        # [300. 300. 300.]
        # [  2.   1.   0.]]

        print(f"second_data {a.data['bounding_box']}")  
        #second_data [[  0.  -1.   0.]
        # [300. 300. 300.]
        # [  2.   1.   0.]]
```

## Links to added or changed documentation pages
 
N/A

## Further Information and Comments

If you're unfamilliar with descriptors, here's a [guide that I found helpful](https://realpython.com/python-descriptors/)

There are quite a few `getter`/`setter` methods that I think should be properties instead. Also, there are some hanging getters that only access the attributes. I left these behind, but they should also probably be deleted since they're unused....

## Reviewer Checklist
- [ ] The PR title is descriptive enough for the changelog, and the PR is labeled correctly
- [ ] If applicable: newly added non-private functions and classes have a docstring including a short summary and a PARAMETERS section
- [ ] If applicable: newly added functions and classes are tested

### CODE REVIEW

File: manim/mobject/opengl_mobject.py [hunk 1]
```diff
@@ -44,6 +45,13 @@ class OpenGLMobject:
         ("point", np.float32, (3,)),
     ]
     shader_folder = ""
+    points = _Data()
```
Comment (ryanmccauley211) [inline manim/mobject/opengl_mobject.py:48]: Why are the class attributes needed as well as the instance ones?

**Review by ryanmccauley211 — CHANGES_REQUESTED** [review]
I like this approach, my only concern with it is that when calling _Data() it isn't obvious at first what it is doing, not sure if there is a way to make it clearer though.

Can you also update opengl_text_mobject.py please? It has a few more using the old notation `each.data["points"]`

https://github.com/ManimCommunity/manim/blob/092555adc787a9c65bfead221c0f88c6a5adbb84/manim/mobject/svg/opengl_text_mobject.py#L477

https://github.com/ManimCommunity/manim/blob/092555adc787a9c65bfead221c0f88c6a5adbb84/manim/mobject/svg/opengl_text_mobject.py#L972

I left a comment as I wasn't sure why both class and instance attributes were being used. If I try calling class ones like `OpenGLSurface.points` I get the error `{AttributeError}'NoneType' object has no attribute '__dict__'`

**Comment by hydrobeam** [comment]
> I like this approach, my only concern with it is that when calling _Data() it isn't obvious at first what it is doing, not sure if there is a way to make it clearer though.

That's a fair, it's a little hard to follow the code currently. I'll write some documentation + comments to make it clearer.

> Can you also update opengl_text_mobject.py please? It has a few more using the old notation each.data["points"]

I initially decided to leave that file alone since it's eventually going to be removed via #1843,  but I've recently turned that one in a draft since I realized there's still more bloat in `mobject.py` that I forgot to remove. I'll get it back up by the end of today. Although if you still think that I should convert the old notation for `opengl_text_mobject.py` here then it'd be fine by me.

If you have other questions/suggestions let me know 👍🙇‍♀️

File: manim/mobject/opengl_mobject.py (hunk 1 above)
Comment (hydrobeam) [reply manim/mobject/opengl_mobject.py:48]: From my understanding, the class variable here tells manim that this is a `_Data` variable and defines how it will later be accessed/ defined. The instance variable in `init_data`  relies on this definition and sets the variable within `self.data` of the particular instance of that mobject i.e. (`Square()`). It might(??) be *technically* possible to only use an instance/class variable? But this is the convention I've seen for descriptors after researching it.

Here's also a guide from the python docs that shows helpful examples:
https://docs.python.org/3/howto/descriptor.html#simple-example-a-descriptor-that-returns-a-constant

https://docs.python.org/3/howto/descriptor.html#managed-attributes

Comment (ryanmccauley211) [reply manim/mobject/opengl_mobject.py:48]: Ah I see, it makes sense now and the implementation looks good in that case

**Comment by ryanmccauley211** [comment]
> I initially decided to leave that file alone since it's eventually going to be removed via #1843, but I've recently turned that one in a draft since I realized there's still more bloat in `mobject.py` that I forgot to remove. I'll get it back up by the end of today. Although if you still think that I should convert the old notation for `opengl_text_mobject.py` here then it'd be fine by me.

Ah ok I didn't realise it was to be removed, in that case don't think there is any need to convert the notation in that file. Rest of the PR looks good to me!

**Review by ryanmccauley211 — APPROVED** [review]
